lesson5/src/MyThread.java

```diff
@@ -0,0 +1,24 @@
+class MyThread extends Thread {
+
+    String name;
+    float arr[];
+    int differ;
+
+
+    MyThread(String name, float[] arr) {
+        this.name = name;
+        this.arr = arr;
+
+  }
+
+    @Override
+    public void run() {
+        if (name == "a1") differ = 0;
+        else if (name == "a2") differ = 5000000;
```

Comment:
Our static analysis engine detects that your pull request includes a change from a Comparison of String objects using `==` or `!=`. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq), unless both strings are either constants in a source file, or have been interned using the `String.intern()` method, the same string value may be represented by two different String objects. Consider using the `equals(Object)` method instead.